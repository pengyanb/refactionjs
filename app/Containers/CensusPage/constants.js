export const ACTION_READ_DATA = "app/Census/ACTION_READ_DATA";
export const ACTION_SET_READ_DATA_RESULT =
  "app/Census/ACTION_SET_READ_DATA_RESULT";
