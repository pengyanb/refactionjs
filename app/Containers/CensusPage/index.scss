.Select-input input {
  background: rgba(0, 0, 0, 0) !important ;
  border: 1px solid rgba(0, 0, 0, 0) !important;
  padding-left: 0 !important;
}
