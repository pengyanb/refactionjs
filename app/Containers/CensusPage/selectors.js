import { createSelector } from "reselect";

const selectCensusPageDomain = () => state => state.get("CensusPage");

const makeSelectCensusPage = () =>
  createSelector(selectCensusPageDomain(), substate => substate.toJS());

export default makeSelectCensusPage;
export { selectCensusPageDomain };
