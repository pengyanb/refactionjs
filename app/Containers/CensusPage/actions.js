import { ACTION_READ_DATA, ACTION_SET_READ_DATA_RESULT } from "./constants";

export function actionReadData(filterInfo) {
  return {
    type: ACTION_READ_DATA,
    filterInfo
  };
}

export function actionSetReadDataResult(result) {
  return {
    type: ACTION_SET_READ_DATA_RESULT,
    result
  };
}
