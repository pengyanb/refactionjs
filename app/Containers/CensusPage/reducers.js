import { fromJS } from "immutable";
import { ACTION_SET_READ_DATA_RESULT } from "./constants";

const initialState = fromJS({
  readDataResult: {}
});

function censusPageReducer(state = initialState, action) {
  switch (action.type) {
    case ACTION_SET_READ_DATA_RESULT:
      return state.set("readDataResult", action.result);
    default:
      return state;
  }
}

export default censusPageReducer;
