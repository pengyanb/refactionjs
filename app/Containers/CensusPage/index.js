import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { compose } from "redux";
import { withRouter } from "react-router-dom";
import { createStructuredSelector } from "reselect";
import ReactSelect from "react-select";
import injectReducer from "../../utils/injectReducer";
import injectSaga from "../../utils/injectSaga";

import reducer from "./reducers";
import saga from "./sagas";
import makeSelectCensusPage from "./selectors";

import { actionReadData } from "./actions";

import DataTableComponent from "../../Components/DataTableComponent";

if (process.env.NODE_ENV !== 'test') {
  require("react-select/dist/react-select.css");
  require("./index.scss");
}

const COLUMN_MAPPER = [
  {
    key: "_id",
    label: "ID",
    filterable: false,
    sortable: true
  },
  {
    key: "name",
    label: "Name",
    filterable: true,
    sortable: true
  },
  {
    key: "gender",
    label: "Gender",
    filterable: true,
    sortable: true,
    useCustomFilter: true
  },
  {
    key: "age",
    label: "Age",
    filterable: true,
    sortable: true,
    useCustomFilter: true
  }
];

const SELECT_OPTIONS_INFO = {
  gender: [
    {
      value: "male",
      label: "male"
    },
    {
      value: "female",
      label: "female"
    }
  ],
  age: [
    {
      value: 20,
      label: "20"
    },
    {
      value: 24,
      label: "24"
    },
    {
      value: 30,
      label: "30"
    },
    {
      value: 55,
      label: "55"
    }
  ]
};
export class CensusPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filterInfo: {
        gender: [],
        age: {
          $gte: null,
          $lte: null
        },
        name: ""
      }
    };
  }
  componentWillMount() {
    this.props.readData(this.state.filterInfo);
  }
  // componentWillReceiveProps(nextProps) {
  //   console.log("[nextProps]: ", nextProps);
  // }
  projectDataTableColumns = () => {
    return COLUMN_MAPPER.map(columnInfo => {
      return {
        id: columnInfo.key,
        Header: () => <h5>{columnInfo.label}</h5>,
        filterable: columnInfo.filterable,
        sortable: columnInfo.sortable,
        getHeaderProps: () => {
          if (columnInfo.useCustomFilter === true) {
            return {
              style: {
                overflow: "visible"
              }
            };
          } else {
            return {};
          }
        },
        Cell: props => {
          const originalData = props.original[columnInfo.key];
          const index = props.index;
          if (typeof originalData !== typeof undefined) {
            return (
              <div style={{ width: "100%", textAlign: "center" }}>
                <span>{originalData}</span>
              </div>
            );
          } else {
            return null;
          }
        }
      };
    });
  };
  render() {
    let data;
    if (this.props.CensusPage.readDataResult.error === false) {
      data = this.props.CensusPage.readDataResult.data;
    }
    return (
      <div>
        {data ? (
          <DataTableComponent
            data={data}
            columns={this.projectDataTableColumns()} 
            FilterComponent={filterComponentInfo => {
              const columnId = filterComponentInfo.column.id;
              let useCustomFilter = false;
              let columnKey;
              COLUMN_MAPPER.map(columnInfo => {
                if (
                  columnInfo.key === columnId &&
                  columnInfo.useCustomFilter === true
                ) {
                  useCustomFilter = true;
                  columnKey = columnInfo.key;
                }
              });
              let value = this.state.filterInfo[columnKey];
              if (useCustomFilter) {
                if (columnKey === "age") {
                  return (
                    <div>
                      <div style={{ display: "inline-block", width: "10%" }}>
                        <span>{">="}</span>
                      </div>
                      <div style={{ display: "inline-block", width: "40%" }}>
                        <ReactSelect
                          value={this.state.filterInfo[columnKey].$gte}
                          options={SELECT_OPTIONS_INFO[columnKey]}
                          clearable
                          placeholder="greater or equal"
                          onChange={selectedItem => {
                            const filterInfo = this.state.filterInfo;
                            if (selectedItem) {
                              filterInfo[columnKey].$gte = selectedItem.value;
                            } else {
                              filterInfo[columnKey].$gte = null;
                            }
                            this.setState({ filterInfo }, () => {
                              this.props.readData(this.state.filterInfo);
                            });
                          }}
                        />
                      </div>
                      <div style={{ display: "inline-block", width: "10%" }}>
                        <span>{"<="}</span>
                      </div>
                      <div style={{ display: "inline-block", width: "40%" }}>
                        <ReactSelect
                          value={this.state.filterInfo[columnKey].$lte}
                          options={SELECT_OPTIONS_INFO[columnKey]}
                          clearable
                          placeholder="less or equal"
                          onChange={selectedItem => {
                            const filterInfo = this.state.filterInfo;
                            if (selectedItem) {
                              filterInfo[columnKey].$lte = selectedItem.value;
                            } else {
                              filterInfo[columnKey].$lte = null;
                            }
                            this.setState({ filterInfo }, () => {
                              this.props.readData(this.state.filterInfo);
                            });
                          }}
                        />
                      </div>
                    </div>
                  );
                } else {
                  return (
                    <div>
                      <ReactSelect
                        value={this.state.filterInfo[columnKey]}
                        options={SELECT_OPTIONS_INFO[columnKey]}
                        searchable
                        clearable
                        multi
                        onChange={selectedItems => {
                          const filterInfo = this.state.filterInfo;
                          if (selectedItems) {
                            filterInfo[columnKey] = selectedItems.map(
                              selectedItem => selectedItem.value
                            );
                          } else {
                            filterInfo[columnKey] = [];
                          }
                          this.setState({ filterInfo }, () => {
                            this.props.readData(this.state.filterInfo);
                          });
                        }}
                      />
                    </div>
                  );
                }
              } else {
                return (
                  <div>
                    <input
                      type="text"
                      onChange={evt => {
                        console.log("evt: ", evt.target.value);
                        const filterInfo = this.state.filterInfo;
                        filterInfo.name = evt.target.value;
                        this.setState(filterInfo, () => {
                          this.props.readData(this.state.filterInfo);
                        });
                      }}
                    />
                  </div>
                );
              }
            }}
          />
        ) : null}
      </div>
    );
  }
}

CensusPage.propTypes = {
  readData: PropTypes.func.isRequired,
  CensusPage: PropTypes.object.isRequired,
};

const mapStateToProps = createStructuredSelector({
  CensusPage: makeSelectCensusPage()
});

const mapDispatchToProps = dispatch => {
  return {
    readData: filterInfo => dispatch(actionReadData(filterInfo))
  };
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: "CensusPage", reducer });
const withSaga = injectSaga({ key: "CensusPage", saga });

export default compose(withReducer, withSaga, withRouter, withConnect)(
  CensusPage
);
