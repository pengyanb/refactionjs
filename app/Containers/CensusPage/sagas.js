import { call, put, takeLatest } from "redux-saga/effects";
import axios from "axios";
import querystring from "querystring";
import { ACTION_READ_DATA } from "./constants";

import { actionSetReadDataResult } from "./actions";

export function* readDataRequest(action) {
  const url = "/data/read";
  try {
    const res = yield axios.post(url, action.filterInfo);
    yield put(actionSetReadDataResult(res.data));
  } catch (err) {
    yield put(
      actionSetReadDataResult({ error: true, message: err.toString() })
    );
  }
}

export function* readDataWatcher() {
  yield takeLatest(ACTION_READ_DATA, readDataRequest);
}

export default function* rootSage() {
  yield [readDataWatcher()];
}
