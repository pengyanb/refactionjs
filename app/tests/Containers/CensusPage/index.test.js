import React from 'react';
import renderer from 'react-test-renderer';

import { CensusPage } from '../../../Containers/CensusPage/index';

describe('Containers.CensusPage.index.test', () => {
  it('render correctly', () => {
    const tree = renderer
      .create(<CensusPage
        readData={() => {}} // mock function
        CensusPage={{
          readDataResult: {
            error: false,
            data: [
              {
                name: 'Jim',
                age: 30,
                gender: 'male',
                _id: 'b3Fshn8F976TZCTg',
              },
              {
                name: 'Bob',
                age: 20,
                gender: 'male',
                _id: 'oqnu2ZnPTebp04bG',
              },
            ],
          },
        }}
      />)
      .toJSON();
    console.log('tree: ', tree);
    expect(tree).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
