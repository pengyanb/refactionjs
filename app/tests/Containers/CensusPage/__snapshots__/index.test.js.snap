// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers.CensusPage.index.test render correctly 1`] = `
<div>
  <div
    className="ReactTable striped -highlight"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
      style={undefined}
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "400px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
          style={undefined}
        >
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              <h5>
                ID
              </h5>
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              <h5>
                Name
              </h5>
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "overflow": "visible",
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              <h5>
                Gender
              </h5>
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "overflow": "visible",
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              <h5>
                Age
              </h5>
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-thead -filters"
        style={
          Object {
            "minWidth": "400px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
          style={undefined}
        >
          <div
            className="rt-th"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          />
          <div
            className="rt-th"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div>
              <input
                onChange={[Function]}
                type="text"
              />
            </div>
          </div>
          <div
            className="rt-th"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "overflow": "visible",
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="Select is-clearable is-searchable Select--multi"
                style={undefined}
              >
                <div
                  className="Select-control"
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={undefined}
                >
                  <span
                    className="Select-multi-value-wrapper"
                    id="react-select-2--value"
                  >
                    <div
                      className="Select-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      className="Select-input"
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-activedescendant="react-select-2--value"
                        aria-describedby={undefined}
                        aria-expanded="false"
                        aria-haspopup="false"
                        aria-label={undefined}
                        aria-labelledby={undefined}
                        aria-owns=""
                        className={undefined}
                        id={undefined}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        role="combobox"
                        style={
                          Object {
                            "boxSizing": "content-box",
                            "width": "5px",
                          }
                        }
                        tabIndex={undefined}
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </span>
                  <span
                    className="Select-arrow-zone"
                    onMouseDown={[Function]}
                  >
                    <span
                      className="Select-arrow"
                      onMouseDown={[Function]}
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="rt-th"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "overflow": "visible",
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "10%",
                  }
                }
              >
                <span>
                  &gt;=
                </span>
              </div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "40%",
                  }
                }
              >
                <div
                  className="Select is-clearable is-searchable Select--single"
                  style={undefined}
                >
                  <div
                    className="Select-control"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={undefined}
                  >
                    <span
                      className="Select-multi-value-wrapper"
                      id="react-select-3--value"
                    >
                      <div
                        className="Select-placeholder"
                      >
                        greater or equal
                      </div>
                      <div
                        className="Select-input"
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-activedescendant="react-select-3--value"
                          aria-describedby={undefined}
                          aria-expanded="false"
                          aria-haspopup="false"
                          aria-label={undefined}
                          aria-labelledby={undefined}
                          aria-owns=""
                          className={undefined}
                          id={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          role="combobox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "5px",
                            }
                          }
                          tabIndex={undefined}
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        >
                          
                        </div>
                      </div>
                    </span>
                    <span
                      className="Select-arrow-zone"
                      onMouseDown={[Function]}
                    >
                      <span
                        className="Select-arrow"
                        onMouseDown={[Function]}
                      />
                    </span>
                  </div>
                </div>
              </div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "10%",
                  }
                }
              >
                <span>
                  &lt;=
                </span>
              </div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "40%",
                  }
                }
              >
                <div
                  className="Select is-clearable is-searchable Select--single"
                  style={undefined}
                >
                  <div
                    className="Select-control"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={undefined}
                  >
                    <span
                      className="Select-multi-value-wrapper"
                      id="react-select-4--value"
                    >
                      <div
                        className="Select-placeholder"
                      >
                        less or equal
                      </div>
                      <div
                        className="Select-input"
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-activedescendant="react-select-4--value"
                          aria-describedby={undefined}
                          aria-expanded="false"
                          aria-haspopup="false"
                          aria-label={undefined}
                          aria-labelledby={undefined}
                          aria-owns=""
                          className={undefined}
                          id={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          role="combobox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "5px",
                            }
                          }
                          tabIndex={undefined}
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        >
                          
                        </div>
                      </div>
                    </span>
                    <span
                      className="Select-arrow-zone"
                      onMouseDown={[Function]}
                    >
                      <span
                        className="Select-arrow"
                        onMouseDown={[Function]}
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "400px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -odd"
            role="row"
            style={undefined}
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  b3Fshn8F976TZCTg
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  Jim
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  male
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  30
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -even"
            role="row"
            style={undefined}
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  oqnu2ZnPTebp04bG
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  Bob
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  male
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <span>
                  20
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pagination-bottom"
    >
      <div
        className="-pagination"
        style={undefined}
      >
        <div
          className="-previous"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          className="-center"
        >
          <span
            className="-pageInfo"
          >
            Page
             
            <div
              className="-pageJump"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="number"
                value={1}
              />
            </div>
             
            of
             
            <span
              className="-totalPages"
            >
              1
            </span>
          </span>
          <span
            className="select-wrap -pageSizeOptions"
          >
            <select
              onChange={[Function]}
              value={5}
            >
              <option
                value={5}
              >
                5
                 
                rows
              </option>
              <option
                value={10}
              >
                10
                 
                rows
              </option>
              <option
                value={20}
              >
                20
                 
                rows
              </option>
              <option
                value={25}
              >
                25
                 
                rows
              </option>
              <option
                value={50}
              >
                50
                 
                rows
              </option>
              <option
                value={100}
              >
                100
                 
                rows
              </option>
            </select>
          </span>
        </div>
        <div
          className="-next"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
